src/ppx_tools/dumpast.ml

```diff
@@ -0,0 +1,362 @@
+
+    method string value = string value
+
+    method float value = float value
+
+    method bool value = bool value
+
+    method record label args = object_ (("type", string label) :: args)
+
+    method other _ = string "serializing other values is not supported yet"
+
+    method nativeint _ = string "serializing nativeint is not supported yet"
+
+    method int64 _ = string "serializing int64 is not supported yet"
+
+    method int32 _ = string "serializing int32 is not supported yet"
+
+    method int value = int value
+
+    method list f = list f
+
+    method option f o =
+      match o with
+      | Some x ->
+        let a = f x in
+        a
+      | None -> Jsonoo.Encode.object_ [ ("type", Jsonoo.Encode.string "None") ]
+
+    method constr label args =
+      match args with
+      | [] -> object_ [ ("type", string label) ]
+      | _ -> object_ (("type", string label) :: args)
+
+    method char value = char value
+
+    method array f arg = list id (Array.to_list arg |> List.map f)
+
+    (*FIXME*)
+    method! open_infos _a { popen_expr; popen_override; popen_loc; _ } =
+      let popen_expr = _a popen_expr in
+      let popen_override = self#override_flag popen_override in
+      let popen_loc = self#location popen_loc in
+      let popen_attributes = null in
+      self#record "open_infos"
+        [ ("popen_expr", popen_expr)
+        ; ("popen_override", popen_override)
+        ; ("popen_loc", popen_loc)
+        ; ("popen_attributes", popen_attributes)
+        ]
+  end
+
+let warn_ast_diff method_name =
+  let (_ : _ Promise.t) =
+    Vscode.Window.showErrorMessage
+      ~message:
+        ("Unable to match the original AST with the reparsed one starting at  "
+       ^ method_name)
+      ()
+  in
+  ()
+
+let reparse_ast =
+  let open Traverse_ast2 in
+  let with_reparse f on_reparse =
+    try f () with
+    | Reparse_error m ->
+      warn_ast_diff m;
+      on_reparse ()
+  in
+  object (self)
+    inherit [Jsonoo.t] lift2 as super
+
+    method unit () = Jsonoo.Encode.null
+
+    method tuple args =
+      Jsonoo.Encode.object_ (("type", Jsonoo.Encode.string "tuple") :: args)
+
+    method string value _ = Jsonoo.Encode.string value
+
+    method float value _ = Jsonoo.Encode.float value
+
+    method bool value _ = Jsonoo.Encode.bool value
+
+    method record label args =
+      Jsonoo.Encode.object_ (("type", Jsonoo.Encode.string label) :: args)
+
+    method other _ _ =
+      Jsonoo.Encode.string "serializing other values is not supported yet"
+
+    method nativeint _ _ =
+      Jsonoo.Encode.string "serializing nativeint is not supported yet"
+
+    method int64 _ _ =
+      Jsonoo.Encode.string "serializing int64 is not supported yet"
+
+    method int32 _ _ =
+      Jsonoo.Encode.string "serializing int32 is not supported yet"
+
+    method int value _ = Jsonoo.Encode.int value
+
+    method list f l l' =
+      (*FIXME different length?*)
+      if List.length l > List.length l' then
```

Comment:
Could you clarify the FIXME?

Also, there's no need to compute the lengths of `l` and `l'` twice.

Reply:
The issue was that reparsing sometimes gave additional (or less, I don't recall) information making lists of different sizes. It is resolved now by always considering the longest list. (So the FIXME still being here is an oversight on my part)